src/target.cpp

```diff
@@ -863,6 +863,10 @@ Buf *target_dynamic_linker(ZigTarget *target) {
             env == ZigLLVM_GNUX32)
     {
         return buf_create_from_str("/libx32/ld-linux-x32.so.2");
+    } else if (arch == ZigLLVM_x86_64 &&
+            (env == ZigLLVM_Musl || env == ZigLLVM_MuslEABI || env == ZigLLVM_MuslEABIHF))
```

Comment:
Correct me if I'm wrong but EABI is ARM-only, isn't it?

Reply:
I thought it stood for ELF Application Binary Interface. But now that you mention it, I don't know how x86_64 could have Hard Float.

I should probably consult the code in clang that this was originally ported from - I'm sure they've added correct handling for the new Musl environment entries.